From feb3416f77ccf1fae976c693bf932318fb370e2a Mon Sep 17 00:00:00 2001
From: Robin Gong <b38343@freescale.com>
Date: Thu, 18 Jun 2015 09:40:58 +0800
Subject: [PATCH 0191/5242] MLK-11122: dma: imx-sdma: use late system sleep pm
 ops

commit  1aad3a412c97edb8584c93322f04cefbfe6b8325 from
https://source.codeaurora.org/external/imx/linux-imx.git

Use SET_LATE_SYSTEM_SLEEP_PM_OPS rather than the common sleep pm ops to ensure
sdma has resumed back before all other module drivers which use sdma resume
back.

Signed-off-by: Robin Gong <b38343@freescale.com>
(cherry picked from commit a7f8725509b494c3073b1bcca63252d5c61bb80d)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/dma/imx-sdma.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/dma/imx-sdma.c b/drivers/dma/imx-sdma.c
index d8ff0fd..9a4a069 100644
--- a/drivers/dma/imx-sdma.c
+++ b/drivers/dma/imx-sdma.c
@@ -2231,7 +2231,7 @@ static int sdma_resume(struct device *dev)
 #endif
 
 static const struct dev_pm_ops sdma_pm_ops = {
-	SET_SYSTEM_SLEEP_PM_OPS(sdma_suspend, sdma_resume)
+	SET_LATE_SYSTEM_SLEEP_PM_OPS(sdma_suspend, sdma_resume)
 };
 
 static struct platform_driver sdma_driver = {
-- 
1.7.9.5

